src/main/java/at/ac/tuwien/kr/alpha/common/atoms/ComparisonLiteral.java

```diff
@@ -125,9 +125,29 @@ public ComparisonLiteral substitute(Substitution substitution) {
 		List<Term> terms = getTerms();
 
 		// Treat case where this is just comparison with all variables bound by partialSubstitution.
-		if (!isLeftAssigning() && !isRightAssigning()) {
-			Term leftEvaluatedSubstitute = substituteAndEvaluate(terms.get(0), partialSubstitution);
-			Term rightEvaluatedSubstitute = substituteAndEvaluate(terms.get(1), partialSubstitution);
+		final Term left = getAtom().getTerms().get(0).substitute(partialSubstitution);
+		final Term right = getAtom().getTerms().get(1).substitute(partialSubstitution);
+		final boolean leftAssigning = assignable(left);
+		final boolean rightAssigning = assignable(right);
+		if (!leftAssigning && !rightAssigning) {
+			// No assignment (variables are bound by partialSubstitution), thus evaluate comparison only.
+			Term leftEvaluatedSubstitute = left;
+			Term rightEvaluatedSubstitute = right;
+			// Evaluate arithmetics.
+			if (left instanceof ArithmeticTerm) {
```

Comment:
Here we have some code duplication for `left` and `right` which could be resolved by introducing a new method.